File: least_count.c
#include <stdio.h>
#include <string.h>
#include <time.h>

int main(void) {
    char sentence[100];  // Increased size to handle longer sentences with spaces
    int number_of_letters = 0;
    
    printf("Enter a sentence: ");
    fgets(sentence, sizeof(sentence), stdin);  // Read entire sentence with spaces
    
    // Remove newline character from the input (if any)
    sentence[strcspn(sentence, "\n")] = 0;

    time_t start, end;
    start = clock();
    
    // Iterate through the sentence and count non-space characters
    for (int i = 0; i < strlen(sentence); i++) {
        if (sentence[i] != ' ') {
            number_of_letters++;
        }
    }

    end = clock();
    
    printf("Execution Time: %ld\n", (end - start));
    printf("\nThe number of letters in the given sentence are: %d\n", number_of_letters);
    
    return 0;
}

/* Observations:
   1. The program uses fgets to handle sentences with spaces.
   2. It correctly removes the newline character that fgets captures.
   3. It counts the number of non-space characters and prints the result.

Example Outputs:
   
Case 1:
Enter a sentence: suraj
The number of letters in the given sentence are: 5

Case 2:
Enter a sentence: suraj aravind
The number of letters in the given sentence are: 12
*/

File: least_count.py
import time

def count_letters(sentence):
    number_of_letters = 0
    for char in sentence:
        if char != ' ':
            number_of_letters += 1
    return number_of_letters

def main():
    sentence = input("Enter a sentence: ")

    start_time = time.time()
    number_of_letters = count_letters(sentence)
    end_time = time.time()

    execution_time = end_time - start_time

    print(f"Execution Time: {execution_time:.6f} seconds")
    print(f"\nThe number of letters in the given sentence are: {number_of_letters}")

if __name__ == "__main__":
    main()
